src/middlewared/middlewared/restful.py

```diff
@@ -18,19 +18,28 @@
 async def authenticate(middleware, req):
 
     auth = req.headers.get('Authorization')
-    if auth is None or not auth.startswith('Basic '):
-        raise web.HTTPUnauthorized()
-    try:
-        username, password = base64.b64decode(auth[6:]).decode('utf8').split(':', 1)
-    except binascii.Error:
+    if auth is None:
         raise web.HTTPUnauthorized()
 
-    try:
-        if not await middleware.call('auth.check_user', username, password):
+    if auth.startswith('Basic '):
+        try:
+            username, password = base64.b64decode(auth[6:]).decode('utf8').split(':', 1)
+        except binascii.Error:
+            raise web.HTTPUnauthorized()
+
+        try:
+            if not await middleware.call('auth.check_user', username, password):
+                raise web.HTTPUnauthorized()
+        except web.HTTPUnauthorized:
+            raise
+        except Exception:
+            raise web.HTTPUnauthorized()
+    elif auth.startswith('API-Key '):
```

Comment:
This is not a valid authentication schema recognized by IANA, is it?

Reply:
Is `Bearer` ok?